mock/mock.go

```diff
@@ -209,6 +209,12 @@ func (m *Mock) On(methodName string, arguments ...interface{}) *Call {
 	return c
 }
 
+// IsCallExpected checks if a method call is expected.
+func (m *Mock) IsCallExpected(method string, arguments ...interface{}) bool {
+	found, _ := m.findExpectedCall(method, arguments...)
+	return found >= 0
+}
```

Comment:
@yiminc you can accomplish what you want without exposing `testify/mock`'s internals.

Something like this:

```go
type TestRuntimeEnvironment struct {
    mock.Mock
    mocked map[string]struct{}
}

func (t *TestRuntimeEnvironment) On(methodName string, arguments ...interface{}) *mock.Call {
    if t.mocked == nil {
        t.mocked = make(map[string]struct{})
    }
    t.mocked[methodName] = struct{}
    return t.Mock.On(methodName, arguments...)
}

func (t *TestRuntimeEnvironment) ExecuteActivity(activityName string, args... interface{}) (result, error) {
    _, hasMock := t.mocked[activityName]
    if hasMock {
        retArgs := t.MethodCalled(activityName, args...)
        // proceed with retArgs
        ...
    } else {
        // no mock found, so proceed with normal code path to call the registered method
        ...
    }
}
```